### ISSUE TITLE
GDScript: Crash when is-operator right-hand-side is null

### ISSUE THREAD

**Issue by nene** [godotengine/godot#59082]
### Godot version

3.4.2.stable

### System information

MacOS 11.6.2

### Issue description

Game crashes without any error message when right-hand-side of `is` operator is `null`.

### Steps to reproduce

Create a scene with a single node and add this script:

```gdscript
extends Node

var MissingClass = null

func _ready():
    self is MissingClass
```

An actual real-life use-case is when you load a script file with incorrect path, which produces `null` (and annoyingly no error or warning whatsoever):

```gdscript
func is_game_item(item):
    var GameItem = load("res://wrong/path/to/GameItem.gd")
    return item is GameItem
```

### Minimal reproduction project

I doubt it being project-specific. I experienced it in one project, then reproduced in a test project with single scene.

**Comment by Calinou** [comment]
I can't reproduce this on `3.x` 7c49b8e70:

![image](https://user-images.githubusercontent.com/180032/158037408-6273686c-711f-4980-85f2-b1f27ebf90d7.png)

**Comment by nene** [comment]
Indeed, Godot 3.4.3 is able to infer that `MissingClass` is `null` and will give error at compile time.

However, when the value can't be determined at compile time, it still crashes. For example with the following code:

```gdscript
extends Node

func _ready():
    var MissingClass = load("res://WrongPath.gd")
    self is MissingClass
```

**Comment by timothyqiu** [comment]
Fixed by #59111.

**Comment by nene** [comment]
Awesome. Thanks!
